src/Persistence/Proxy/PersistedObjectsTracker.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the zenstruck/foundry package.
+ *
+ * (c) Kevin Bond <kevinbond@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Zenstruck\Foundry\Persistence\Proxy;
+
+use Zenstruck\Foundry\Configuration;
+
+/**
+ * @internal
+ */
+final class PersistedObjectsTracker
+{
+    /**
+     * This buffer of objects needs to be static to be kept between two kernel.reset events.
+     *
+     * @var list<\WeakReference<object>>
+     */
+    private static $buffer = [];
+
+    public function refresh(): void
+    {
+        self::proxifyObjects();
+    }
```

Comment:
`proxifyObjects()` being now a private method, this non-static method is the only entrypoint to trigger the refresh